Library/brew.rb

```diff
@@ -39,13 +38,24 @@
   EOS
 end
 
+# Check for user agreement of the Xcode license before permitting
+# any other brew usage to continue. This prevents the situation where
+# people are instructed to "please re-run as root via sudo" on brew commands.
+# The check can only fail when Xcode is installed & the active developer dir.
+if `/usr/bin/xcrun clang 2>&1` =~ /license/ && !$?.success?
+  odie <<-EOS.undent
+    You have not agreed to the Xcode license. Please resolve this by running:
+      sudo xcodebuild -license
+  EOS
+end
```

Comment:
> Does that happen with all commands or just doctor? 

You can run `info`, `desc`, `audit` (but not `style`), `cat`, `cleanup`, `create`, `edit`, `deps`, `fetch`, `irb`, `leaves`, `link`, `linkapps`, `list`, `prune`, `readall`, `pin` and `unpin`, `search` and `uses`.

`install`, `doctor` both fail, so does `update` (very loudly), `upgrade` fails, as does `tap`, as does `test-bot` relatively obviously, `pull` fails messily when it throws the license message & fails to apply the patch. `brew tests` breaks trying to compile the gems with the license failure.  

`config` partially fails:
```
Xcode: 7.0.1
CLT: N/A
Agreeing to the Xcode/iOS license requires admin privileges, please re-run as root via sudo.
Clang: N/A
X11: N/A
System Ruby: 2.0.0-p645
```
`gist-logs` partially fails:
```
Vagrants-Mac:~ vagrant$ brew gist-logs pkg-config
Agreeing to the Xcode/iOS license requires admin privileges, please re-run as root via sudo.
https://gist.github.com/2382bae6d49a1491be4d
```